Makefile

```diff
@@ -94,7 +95,9 @@ K_JAR := $(K_SUBMODULE)/k-distribution/target/release/k/lib/java/kernel-1.0-SNAP
 deps: repo-deps
 repo-deps: k-deps plugin-deps
 k-deps: $(K_JAR)
-plugin-deps: $(PLUGIN_SUBMODULE)/client-c/main.cpp
+plugin-deps: $(PLUGIN_SUBMODULE)/client-c/main.cpp $(wildcard $(PLUGIN_SUBMODULE)/plugin/*)
```

Comment:
Why do you need to add the `$(wildcard ...)`? At best it is depending on too many things. At worst, those files you're trying to pick up won't exist but Makefile will silently ignore the fact that they don't exist. If there are specific files you need to be present, list them here.

Alternatively, you can just remove all dependencies for this top-level `plugin-deps`.

Reply:
I'm in favor of either being precise or not worrying about it. Since being precise becomes more complicated over time, I have removed explicit dependencies from this target.